### PR TITLE
Correctly display dirty and deleted nodes in flat lists

### PR DESCRIPTION
**Pull request by danielkestler**
bildschirmfoto 2018-10-12 um 12 33 31

### CODE REVIEW

File: Classes/Controller/StandardController.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function queryAction($preset, $nodeContextPath, $page = 1)
         ];
         $nodes = \Neos\Eel\Utility::evaluateEelExpression($expression, $this->eelEvaluator, $contextVariables, $this->defaultContextConfiguration);
         $nodeInfoHelper = new NodeInfoHelper();
-        $result = $nodeInfoHelper->renderNodes($nodes, $this->getControllerContext(), true);
+        $result = $nodeInfoHelper->renderNodes($nodes, $this->getControllerContext(), false);
```
Comment (dimaip) [inline Classes/Controller/StandardController.php]: Why is this necessary? It will make things A LOT slower...

Comment (danielkestler) [reply Classes/Controller/StandardController.php]: Without it, we don't know if the node is deleted ... I don't know why it is returned anyway if deleted, there must be some bug in the Eel.

Comment (danielkestler) [reply Classes/Controller/StandardController.php]: Okay, I will have another look ;-)

**Comment by danielkestler** [comment]
Do you like it more this way?

**Comment by danielkestler** [comment]
The latest commit also displays hidden nodes correctly, see screenshot:
bildschirmfoto 2018-10-12 um 16 36 03

**Comment by danielkestler** [comment]
Now includes a bugfix for "delayed" showing of isHidden-state of nodes and also displays hiddenBefore and hiddenAfter states correctly, see screenshot:

bildschirmfoto 2018-10-15 um 10 28 19

**Comment by dimaip** [comment]
Sorry, but I still don't understand why it's necessary to filter out removed nodes. Shouldn't they not be returned in the first place?

**Comment by danielkestler** [comment]
Yes, of course, but they are. I don't know where the bug occurs ...

But it makes sense to display the deleted, but not yet published nodes in the tree. See this PR in Neos.Ui: https://github.com/neos/neos-ui/pull/1929

**Comment by dimaip** [comment]
Ah, ok, I see.

**Comment by dimaip** [comment]
Oups, this PR flew out of my mind completely! Could you please rebase & rebuild and I'd merge?

**Comment by danielkestler** [comment]
I hope I did it right ;-) Changes are merged.

File: Resources/Private/FlatNav/src/FlatNav.js [hunk 2]
```diff
@@ -81,29 +82,72 @@ export default class FlatNav extends Component {
...
                 const item = $get([contextPath], this.props.nodeData);
+
                 if (item) {
-                    const nodeTypeName = $get('nodeType', item);
-                    const nodeType = this.props.nodeTypesRegistry.getNodeType(nodeTypeName);
+                    const isFocused = this.props.focused === contextPath;
+                    const isDirty = this.props.publishableNodes.filter(i => (
+                        $get('contextPath', i) === contextPath ||
+                        $get('documentContextPath', i) === contextPath
+                    )).count() > 0;
```
Comment (pmaas) [inline Resources/Private/FlatNav/src/FlatNav.js]: This should be `length` instead of `count()`, right?

Comment (pmaas) [reply Resources/Private/FlatNav/src/FlatNav.js]: Fixed it here: https://github.com/psmb/Psmb.FlatNav/pull/20
